Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSTubeCalibration.py

```diff
@@ -0,0 +1,1078 @@
+        scale_alg.setProperty("OutputWorkspace", scaled_ws_name)
+        scale_alg.execute()
+
+        return mtd[scaled_ws_name]
+
+    def _merge_workspaces(self, ws_to_known_edges):
+        """
+        Merge the workspaces containing the individual strips into a single workspace containing all the strips.
+
+        :param ws_to_known_edges: a dictionary of workspaces paired with their known strip edge locations.
+        """
+
+        # Find the boundaries that isolate the strips in each calibration dataset
+        ws_to_boundaries = self._get_boundaries_for_each_strip(ws_to_known_edges)
+
+        # This step allows us to multiply all the workspaces together in order to merge them into a single dataset.
+        for ws, boundaries in ws_to_boundaries.items():
+            self.log().debug(f"Isolating shadow in {ws} between boundaries {boundaries[0]} and {boundaries[1]}.")
+            self._set_counts_to_one_outside_strip_boundaries(ws, boundaries)
+
+        # Perform the merge to get a single workspace containing all the strips
+        self._multiply_ws_list(list(ws_to_boundaries.keys()))
+
+    def _multiply_ws_list(self, ws_list):
+        self.log().debug("Multiplying workspaces together...")
+        rhs_ws = ws_list[0]
+
+        if len(ws_list) == 1:
+            # If there is only one workspace in the list then there is nothing to multiply together.
+            # We re-name the workspace because we still need a workspace matching the output ws name but
+            # there is no need to also keep the scaled workspace in this situation.
+            self._rename_workspace(rhs_ws, self._MERGED_WS_NAME)
+            return
+
+        for ws in ws_list[1:]:
+            alg = self._create_child_alg("Multiply", RHSWorkspace=rhs_ws, LHSWorkspace=ws)
+            alg.execute()
+            rhs_ws = alg.getProperty("OutputWorkspace").value
+
+        AnalysisDataService.addOrReplace(self._MERGED_WS_NAME, rhs_ws)
+
+    def _get_tube_name(self, tube_id):
+        """Construct the name of the tube based on the id given"""
+        side = TubeSide.get_tube_side(tube_id)
+        tube_side_num = tube_id // 2  # Round down to get integer name for tube
+        return self._detector_name + "-detector/" + side + str(tube_side_num)
+
+    def _get_tube_data(self, tube_id, ws):
+        """Return an array of all counts for the given tube"""
+        tube_name = self._get_tube_name(tube_id)
+
+        # The TubeSpec class has been used to make this easier than interrogating the IDF
+        tube_spec = TubeSpec(ws)
+        tube_spec.setTubeSpecByString(tube_name)
+        if not tube_spec.getNumTubes() == 1:
+            raise RuntimeError(f"Found more than one tube for tube id {tube_id}")
+        tube_ws_index_list = tube_spec.getTube(0)[0]
+        if not len(tube_ws_index_list) == DetectorInfo.NUM_PIXELS_IN_TUBE:
+            raise RuntimeError(f"Found incorrect number of counts for tube id {tube_id}")
```

Comment:
Good point, lets add something to the maintenance issue to review this error message and make that sure it's describing the right thing. This step is looking up the specific workspace indices for the tube so that we can grab only the data for that detector tube from the workspace. Something like " Found incorrect number of workspace indices" may be the most accurate description?

Reply:
I've added something for this to the maintenance issue - #35982